### PR TITLE
[Tune] Fix PBT hyperparameter mutation bug when config value is int type with Float distribution

### LINKED ISSUE #54755: [Tune] PBT scheduler ignoring some of ``hyperparam_mutations`` after restore
### What happened + What you expected to happen

When using ``PopulationBasedTraining`` and restoring from a checkpoint, some of the original ``hyperparameter_mutations`` seem to be ignored, resulting in resampling returning the same value.

In the following example, ``clip_param`` seems to be set to 0. Thus, no gradients are actually being optimized, apart for the specific case when the ratio in the PPO's objective is below 1. 

```
lr : 0.0006183230705195503 --- (* 0.8) --> 0.0004946584564156402
num_sgd_iter : 1 --- (* 0.8) --> 1
entropy_coeff : 0.0005671363125642761 --- (resample) --> 0.0009495159664290651
clip_param : 0 --- (resample) --> 0
lambda : 0.9683199949528589 --- (* 0.8) --> 0.7746559959622872
```

The parameter of hyperparameter mutations is as follows:
```
hyperparam_mutations={  
    "lr": tune.uniform(1e-5, 1e-3),
    "clip_param": tune.uniform(0.005, 0.5),
    "entropy_coeff": tune.uniform(0.0, 0.002),
    "lambda": tune.uniform(0.5, 1.0),
    "num_sgd_iter": tune.randint(3, 15),
},
```

### Versions / Dependencies

- Ray 2.10.0
- Python 3.11.11

### Reproduction script

```py
PATH = "/root/ray_results/PPOWithCurriculum_2025-07-15_14-44-17"

os.environ["TUNE_DISABLE_AUTO_CALLBACK_LOGGERS"] = "1"  # Don't create result.json and progress.csv
if PATH:
    tuner = tune.Tuner.restore(PATH, PPOWithCurriculum, resume_errored=True, resume_unfinished=True)
else:
    tuner = tune.Tuner(
        "PPO",
        param_space=config,
        run_config=train.RunConfig(
            checkpoint_config=train.CheckpointConfig(10),
            callbacks=[TBXReducedLoggerCallback()],
            # stop={"timesteps_total": 20e6}
        ),
        tune_config=tune.TuneConfig(
            metric="episode_reward_mean",
            mode="max",
            trial_dirname_creator=lambda trial: str(next(cnt)),
            num_samples=20,
            scheduler=PopulationBasedTraining(
                # perturbation_interval=10 * 60,
                perturbation_interval=100,
                time_attr="training_iteration",
                hyperparam_mutations={  
                    "lr": tune.uniform(1e-5, 1e-3),
                    "clip_param": tune.uniform(0.005, 0.5),
                    "entropy_coeff": tune.uniform(0.0, 0.002),
                    "lambda": tune.uniform(0.5, 1.0),
                    "num_sgd_iter": tune.randint(3, 15),
                },
                custom_explore_fn=check_parameters,
            ),
        ),
    )

results = tuner.fit()
```

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by sjp611**
## Description

Fix PBT scheduler's `_explore` function incorrectly converting resampled float values back to integers when the original config value happens to be an `int` type (e.g., `0` instead of `0.0`).

This commonly occurs after JSON serialization/deserialization during checkpoint save/restore, where `0.0` becomes `0`.

**Before fix:** `clip_param: 0 -> 0` (int(0.25) = 0)
**After fix:** `clip_param: 0 -> 0.25`

## Related issues

Fixes #54755

### Root Cause
In `_explore()` function, the type preservation logic converts sampled floats back to int when `isinstance(config[key], int)` is True, regardless of the distribution type.

### Solution
Added check to skip int conversion when distribution is `Float` type:
```python
if isinstance(config[key], int) and not isinstance(distribution, Float):
    new_config[key] = int(new_config[key])

### CODE REVIEW

**Review by TimothySeah — APPROVED** [review]

**Comment by sjp611** [comment]
The premerge CI failure is due to a wanda binary download issue (exit code 22), unrelated to this PR. All relevant checks (lint, tune tests) passed. Could a maintainer please re-trigger the premerge check? Thank you!

**Comment by sjp611** [comment]
CI failure (wanda binary download) is unrelated to this PR.

**Comment by sjp611** [comment]
@justinvyu  Could you please merge this?
